### ISSUE TITLE
RxJs lastValueFrom scheduled asapScheduler strange behaviour

### ISSUE THREAD

**Issue by nigrosimone** [ReactiveX/rxjs#7241]
### Describe the bug

I have this stript for play with `scheduled`/`asapScheduler` but I found a  strange behaviour in combination with `lastValueFrom`:

```ts
    import { of, lastValueFrom, scheduled, asapScheduler } from 'rxjs';
    
    const bbj = { foo: 'bar' };
    
    async function foo() {
      const result = await lastValueFrom(scheduled(of(bbj), asapScheduler));
      console.log(result);
      return result;
    }
    
    async function main() {
      await foo(); // 1° console log
      await foo(); // 2° console log
      await foo(); // 3° console log
    }
    
    document.getElementById('button').addEventListener('click', function () {
      main();
    });
```

the expected behaviour is on button click, 3 console log, eg.:

    {foo: "bar"}
    {foo: "bar"}
    {foo: "bar"}

But on first click it produce NO console log

On second click it produce 5 console log:

    {foo: "bar"}
    {foo: "bar"}
    {foo: "bar"}
    {foo: "bar"}
    {foo: "bar"}

and on third click, 1 console log:

    {foo: "bar"}

Why not produce 3 console log on each click?

### Expected behavior

3 console log on each click

    {foo: "bar"}
    {foo: "bar"}
    {foo: "bar"}

### Reproduction code

```ts
import { of, lastValueFrom, scheduled, asapScheduler } from 'rxjs';
    
    const bbj = { foo: 'bar' };
    
    async function foo() {
      const result = await lastValueFrom(scheduled(of(bbj), asapScheduler));
      console.log(result);
      return result;
    }
    
    async function main() {
      await foo(); // 1° console log
      await foo(); // 2° console log
      await foo(); // 3° console log
    }
    
    document.getElementById('button').addEventListener('click', function () {
      main();
    });
```

### Reproduction URL

try online https://stackblitz.com/edit/rxjs-mkgftu?devtoolsheight=60&file=index.ts

### Version

7

**Comment by voliva** [comment]
It could be an issue on `scheduled` with synchronous values. It looks like for some reason on the first click the observable emits the value, but it doesn't complete the observable.

https://stackblitz.com/edit/rxjs-cwwznh?devtoolsheight=60&file=index.ts

**Comment by voliva** [comment]
I think this was fixed by #7198 

I've bisected it by adding this test:
```ts
it('completes', (done) => {
    const input = of('a');
    scheduled(input, asapScheduler).subscribe({
      complete: done
    });
  });
```
on master and 7.x branch it's fixed, and before that merge it breaks.

**Comment by benlesh** [comment]
This is confirmed fixed in the latest published version of RxJS (7.8.1)
